glue_vispy_viewers/common/vispy_widget.py

```diff
@@ -77,6 +114,20 @@ def __init__(self, parent=None):
         add_callback(self, 'visible_axes', nonpartial(self._toggle_axes))
         add_callback(self, 'perspective_view', nonpartial(self._toggle_perspective))
 
+    # TODO: how to get data here?
+    def update_axis_label(self, data):
+        # data object
+        label = []
+        tick_value = []
+        if data.coords:
+            # TODO: think about 4th dim
+            for i in range(data.ndim):
+                label.append(data.coords.axis_label(i))  # ['Vopt', 'Declination', 'Right Ascension']
+                tick_value.append(data._world_component_ids[label[i]])
+            self.xax.domain = tick_value[0]
+            self.yax.domain = tick_value[1]
+            self.zax.domain = tick_value[2]
+
```

Comment:
I'll take a look at this today :) I was having laptop issues, so didn't manage to get to it yesterday.